arch/x86/core/ia32/swap.S

```diff
@@ -65,27 +69,15 @@
  * unsigned int arch_swap (unsigned int eflags);
  */
 
-.macro read_tsc var_name
-	push %eax
-	push %edx
-	rdtsc
-	mov %eax,\var_name
-	mov %edx,\var_name+4
-	pop %edx
-	pop %eax
-.endm
 SECTION_FUNC(TEXT, arch_swap)
 #ifdef CONFIG_EXECUTION_BENCHMARKING
-	/* Save the eax and edx registers before reading the time stamp
-	* once done pop the values.
-	*/
-	push %eax
-	push %edx
-	rdtsc
-	mov %eax,arch_timing_swap_start
-	mov %edx,arch_timing_swap_start+4
-	pop %edx
-	pop %eax
+	push	%eax
```

Comment:
I know the old code was being used as a template, but it was written by someone who wasn't the sharpest tool in the shed.

the 32-bit SYS V ABI passes in arguments via the stack, so unless I'm missing something you don't need to preserve either eax or edx as neither of these caller-saved registers are used here